test/compile/data/bin.test.ts

```diff
@@ -102,4 +103,64 @@ describe('compile/data/bin', function() {
       expr: 'format(datum["bin_maxbins_10_Rotten_Tomatoes_Rating_start"], \'s\') + \' - \' + format(datum["bin_maxbins_10_Rotten_Tomatoes_Rating_end"], \'s\')'
     });
   });
+
+  it ('should add bin transform and correctly apply bin for binned field without custom extent', () => {
+    const model = parseUnitModel({
+      mark: 'point',
+      encoding: {
+        y: {
+          bin: true,
+          'field': 'Acceleration',
+          'type': 'quantitative'
+        }
+      }
+    });
+    const transform = assemble(model);
+    assert.deepEqual(transform.length, 2);
+    assert.deepEqual<VgTransform>(transform[0], {
+      type: 'extent',
+      field: 'Acceleration',
+      signal: 'bin_maxbins_10_Acceleration_extent'
+    });
+    assert.deepEqual<VgTransform>(transform[1], {
+      type: 'bin',
+      field: 'Acceleration',
+      as: ['bin_maxbins_10_Acceleration_start', 'bin_maxbins_10_Acceleration_end'],
+      maxbins: 10,
+      signal: 'bin_maxbins_10_Acceleration_bins',
+      extent: {signal: 'bin_maxbins_10_Acceleration_extent'}
+    });
+  });
+
+  it('should apply the bin transform if specified BinTransform', () => {
+    const t: BinTransform = {bin: {"maxbins" : 10}, field: "Rotten_Tomatoes_Rating", as: "something"};
+    const model = parseUnitModel({
+      data: {url: "data/movies.json"},
+      mark: "circle",
+      transform: [t],
+      encoding: {
```

Comment:
Not sure why you need to add x and color here -- in any case this is a funny spec because you don't really use the output field from `BinTransform`.